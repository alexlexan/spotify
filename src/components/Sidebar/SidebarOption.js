import React from "react";
import "./SidebarOption.sass";
import { NavLink } from "react-router-dom";
import { useDispatch } from "react-redux";
import { setActivePlaylist } from "../../store/actions";

function SidebarOption({ option = "test", Icon, id, playlist }) {
  const dispatch = useDispatch();
  const getPlaylists = (item) => {
    dispatch(setActivePlaylist(item));
  };

  return (
    <div className="sidebarOption">
      {Icon && <Icon className="sidebarOption__icon" />}
      {Icon ? (
        <h4>{option}</h4>
      ) : (
        <NavLink
          to={`/playlists/${id}/sidebarPlaylist`}
          activeClassName="sidebarOptionActive"
          onClick={() => getPlaylists(playlist)}
        >
          <p>{option}</p>
        </NavLink>
      )}
    </div>
  );
}

export default SidebarOption;
