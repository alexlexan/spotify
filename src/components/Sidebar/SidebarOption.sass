.sidebarOption 
  display: flex
  cursor: pointer
  align-items: center
  height: 40px
  color: grey
  transition: 200ms color ease-in
  a 
    text-decoration: none
    color: grey
    padding: 0 12px
    transition: 200ms color ease-in
    &:hover
      color: white
      transition: 200ms color ease-in

.sidebarOption:hover 
  color: white

.sidebarOptionActive
  color: white    


.sidebar > hr 
  border: 1px solid #282828
  width: 90%
  margin: 10px auto


.sidebarOption__icon 
  padding-left: 10px
  padding-right: 10px


.sidebarOption > p 
  margin-top: 10px
  margin-left: 20px
  font-size: 14px
