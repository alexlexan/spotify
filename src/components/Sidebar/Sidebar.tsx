import React from "react";
import "./Sidebar.sass";
import SidebarOption from "./SidebarOption";
import HomeIcon from "@material-ui/icons/Home";
import SearchIcon from "@material-ui/icons/Search";
import LibraryMusicIcon from "@material-ui/icons/LibraryMusic";
import { NavLink } from "react-router-dom";

const Sidebar: React.FC = () => {
  return (
    <div className="sidebar">
      <img
        className="sidebar__logo"
        src="https://getheavy.com/wp-content/uploads/2019/12/spotify2019-830x350.jpg"
        alt=""
      />
      <NavLink to={`/`} activeClassName="sidebarOptionActive">
        <SidebarOption Icon={HomeIcon} option="Home" />
      </NavLink>

      <NavLink to={`/search`} activeClassName="sidebarOptionActive">
        <SidebarOption Icon={SearchIcon} option="Search" />
      </NavLink>

      <NavLink to={`/myplaylists`} activeClassName="sidebarOptionActive">
        <SidebarOption Icon={LibraryMusicIcon} option="Your Library" />
      </NavLink>
    </div>
  );
};

export default Sidebar;
