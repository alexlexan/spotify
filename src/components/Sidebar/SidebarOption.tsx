import { SvgIconTypeMap } from "@material-ui/core";
import { OverridableComponent } from "@material-ui/core/OverridableComponent";
import React from "react";
import "./SidebarOption.sass";

type Props = {
  option: string;
  Icon: OverridableComponent<SvgIconTypeMap<{}, "svg">>;
};

const SidebarOption: React.FC<Props> = ({ option = "test", Icon }) => {
  return (
    <div className="sidebarOption">
      <Icon className="sidebarOption__icon" />
      <h4>{option}</h4>
    </div>
  );
};

export default SidebarOption;
