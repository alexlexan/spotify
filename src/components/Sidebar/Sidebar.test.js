import React from "react";
import Sidebar from "./Sidebar.tsx";

test("render Sidebar component", () => {
  const component = shallow(<Sidebar />);
  expect(component).toMatchSnapshot();
});
