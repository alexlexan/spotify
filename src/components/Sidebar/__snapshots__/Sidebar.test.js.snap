// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Sidebar component 1`] = `
<div
  className="sidebar"
>
  <img
    alt=""
    className="sidebar__logo"
    src="https://getheavy.com/wp-content/uploads/2019/12/spotify2019-830x350.jpg"
  />
  <NavLink
    activeClassName="sidebarOptionActive"
    to="/"
  >
    <SidebarOption
      Icon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "displayName": "HomeIcon",
          "muiName": "SvgIcon",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
      option="Home"
    />
  </NavLink>
  <NavLink
    activeClassName="sidebarOptionActive"
    to="/search"
  >
    <SidebarOption
      Icon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "displayName": "SearchIcon",
          "muiName": "SvgIcon",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
      option="Search"
    />
  </NavLink>
  <NavLink
    activeClassName="sidebarOptionActive"
    to="/myplaylists"
  >
    <SidebarOption
      Icon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "displayName": "LibraryMusicIcon",
          "muiName": "SvgIcon",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
      option="Your Library"
    />
  </NavLink>
</div>
`;
